### PR TITLE
[frontend] Bulk entity creation of SDO/SCO with Copy/Pasting (#4352)

### PR DESCRIPTION
**Pull request by ValentinBouzinFiligran**
### Proposed changes

* Bulk relationship components on StixCoreRelationshipCreation component
* main relationship selected
* selected element added on popup
* ListLines removed to DataTable
* Custom event listener to refresh knowledge pages
*  representative updated on entity type change

### Related issues

* #4352 

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

**Review by lndrtrbn — APPROVED** [review]

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 1]
```diff
@@ -738,14 +738,15 @@ export const removeIdAndIncorrectKeysFromFilterGroupObject = (filters: FilterGro
...
+export const useBuildEntityTypeBasedFilterContext = (entityTypeParam: string | string[], filters: FilterGroup | undefined): FilterGroup => {
+  const entityType = Array.isArray(entityTypeParam) ? entityTypeParam : [entityTypeParam];
+  const userFilters = useRemoveIdAndIncorrectKeysFromFilterGroupObject(filters, entityType);
   return {
     mode: 'and',
     filters: [
       {
         key: 'entity_type',
-        values: [entityType],
+        values: entityType,
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: minor comment : maybe rename `entityType` to `entityTypes` if it's always an array ?

File: opencti-platform/opencti-front/src/utils/hooks/useLocalStorage.ts [hunk 2]
```diff
@@ -665,16 +665,14 @@ export const usePaginationLocalStorage = <U>(
...
-        const { filters } = viewStorage;
-        const newValue = {
-          ...viewStorage,
-          filters: handleAddFilterWithEmptyValueUtil({ filters, filter }),
-          latestAddFilterId: filter.id,
-        };
-        setValue(newValue);
-        dispatch(`${key}_paginationStorage`, newValue);
-      }
+      const { filters } = viewStorage;
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/utils/hooks/useLocalStorage.ts]: why this change of logic ? just to understand, since it's a function used everywhere.

File: opencti-platform/opencti-front/src/private/components/threats/intrusion_sets/Root.tsx [hunk 3]
```diff
@@ -224,10 +214,10 @@ const RootIntrusionSet = ({ intrusionSetId, queryRef }: RootIntrusionSetProps) =
               <Route
                 path="/knowledge/*"
                 element={
-                  <div data-testid="instrusionSet-knowledge" key={forceUpdate}>
+                  <div key={forceUpdate}>
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/threats/intrusion_sets/Root.tsx:220]: we don't need anymore the data-testid instrusionSet-knowledge ?

File: opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreationFromEntityStixCoreObjectsLine.jsx [hunk 4]
```diff
@@ -139,195 +139,197 @@ const StixCoreRelationshipCreationFromEntityStixCoreObjectsLineComponent = ({
...
 
+export const stixCoreRelationshipCreationFromEntityStixCoreObjectsLineFragment = graphql`
+  fragment StixCoreRelationshipCreationFromEntityStixCoreObjectsLine_node on StixCoreObject {
+    id
+    standard_id
+    parent_types
+    entity_type
+    created_at
+    ... on AttackPattern {
+      name
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreationFromEntityStixCoreObjectsLine.jsx]: I wonder if we could simplify this fragment by using representatives

File: opencti-platform/opencti-front/src/utils/hooks/useLocalStorage.ts (hunk 2 above)
Comment (ValentinBouzinFiligran) [reply opencti-platform/opencti-front/src/utils/hooks/useLocalStorage.ts]: Adrien has corrected this
it was blocking us because without a default filter value, it is completely blocked

File: opencti-platform/opencti-front/src/private/components/threats/intrusion_sets/Root.tsx (hunk 3 above)
Comment (ValentinBouzinFiligran) [reply opencti-platform/opencti-front/src/private/components/threats/intrusion_sets/Root.tsx:220]: i added this few weeks ago but i don't need it anymore

File: opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreationFromEntityStixCoreObjectsLine.jsx (hunk 4 above)
Comment (ValentinBouzinFiligran) [reply opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreationFromEntityStixCoreObjectsLine.jsx]: useToggleEntity and shift entity are based on name
I risk breaking a lot of things if I change to this exact location

File: opencti-platform/opencti-front/src/utils/hooks/useLocalStorage.ts (hunk 2 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/utils/hooks/useLocalStorage.ts]: Yes, the whole point was to have consistency in that function : the goal is yo create an empty filter value but we assume we already have filters in our storage. 

So the change is simply to handle empty value in storage by the using the emptyFilterGroup constant (that we use for example in clearAllFilters)

**Review by aHenryJard — APPROVED** [review]

**Comment by aHenryJard** [comment]
All good for me !
